draft-barnes-acme.md

```diff
@@ -1212,66 +1212,74 @@ If all of the above verifications succeed, then the validation is successful.  I
 
 The Domain Validation with Server Name Indication (DVSNI) validation method aims to ensure that the ACME client has administrative access to the web server at the domain name being validated, and possession of the private key being authorized.  The ACME server verifies that the operator can reconfigure the web server by having the client create a new self-signed challenge certificate and respond to TLS connections from the ACME server with it.
 
-The challenge proceeds as follows: The ACME server sends the client a random value R and a nonce used to identify the transaction.  The client responds with another random value S.  The server initiates a TLS connection on port 443 to one or more of the IPv4 or IPv6 hosts with the domain name being validated.  In the handshake, the ACME server sets the Server Name Indication extension set to "\<nonce\>.acme.invalid".  The TLS server (i.e., the ACME client) should respond with a valid self-signed certificate containing both the domain name being validated and the domain name "\<Z\>.acme.invalid", where Z = SHA-256(R &#124;&#124; S).
+The challenge proceeds as follows: The ACME server sends the client a random value R, a nonce used to identify the transaction, and an array of acceptable ports for which the challenge may be performed.  The client responds with the desired port and key used within the test certificate.  The server initiates a TLS connection on the client's chosen port to one or more of the IPv4 or IPv6 hosts with the domain name being validated.  In the handshake, the ACME server sets the Server Name Indication extension set to "\<nonce\>.acme.invalid".  The TLS server (i.e., the ACME client) should respond with a valid self-signed certificate containing the domain name "\<R\>[0:32].\<R\>[32:64].acme.invalid".
```

Comment:
Thanks, I think this will be more clear after I restructure the challenge towards the form of #170 as suggested by @martinthomson. 

The server's certificate contains the "token domain name" (\<R\>.acme.invalid).  It does not need to contain the domain name intended for authorization.  The "token domain name" is meant to directly tie the individual ACME challenge to the certificate.